diff -Naur kdelibs-4.11.2.orig/CMakeLists.txt kdelibs-4.11.2/CMakeLists.txt
--- kdelibs-4.11.2.orig/CMakeLists.txt	2013-09-26 22:01:41.396214518 +0200
+++ kdelibs-4.11.2/CMakeLists.txt	2014-01-25 01:15:49.536201825 +0100
@@ -85,7 +85,7 @@
     set (STRIGI_TYPE "OPTIONAL")
     set (STRIGI_EXTRA_TEXT "")
 else (WIN32)
-    set (STRIGI_TYPE "REQUIRED")
+    set (STRIGI_TYPE "OPTIONAL")
     set (STRIGI_EXTRA_TEXT "Required by some critical kioslaves")
 endif (WIN32)
 set_package_properties(Strigi PROPERTIES DESCRIPTION "Desktop indexing and search support"
diff -Naur kdelibs-4.11.2.orig/kio/kio/kfilemetainfo.cpp kdelibs-4.11.2/kio/kio/kfilemetainfo.cpp
--- kdelibs-4.11.2.orig/kio/kio/kfilemetainfo.cpp	2013-06-28 19:03:41.365391370 +0200
+++ kdelibs-4.11.2/kio/kio/kfilemetainfo.cpp	2014-01-25 01:11:58.524089874 +0100
@@ -421,9 +421,10 @@
 {
 }
 
-const KFileMetaInfo& KFileMetaInfo::operator= ( KFileMetaInfo const & kfmi )
+KFileMetaInfo &KFileMetaInfo::operator= ( KFileMetaInfo const &kfmi )
 {
-    return kfmi;
+    d = kfmi.d;
+    return *this;
 }
 
 KFileMetaInfo::~KFileMetaInfo()