From 7860feb6ed1c8c2c60945a5c0b33595077228561 Mon Sep 17 00:00:00 2001
From: Lionel Elie Mamane <lionel@mamane.lu>
Date: Thu, 11 Sep 2014 09:39:47 +0200
Subject: system-gltf should follow system-libs

Change-Id: I123b6dd56c21cdecd218152396f8cb8366ef53b2

diff --git a/configure.ac b/configure.ac
index 561a15b..d8bcb80 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1608,7 +1608,7 @@ AC_ARG_WITH(system-jpeg,
 AC_ARG_WITH(system-libgltf,
     AS_HELP_STRING([--with-system-libgltf],
         [Use libgltf already on system.]),,
-    [with_system_libgltf=auto])
+    [with_system_libgltf="$with_system_libs"])
 
 AC_ARG_WITH(system-clucene,
     AS_HELP_STRING([--with-system-clucene],
-- 
cgit v0.10.2
