From 1ea25036f12b4478f7b0e26781913833c1e75049 Mon Sep 17 00:00:00 2001
From: Luke Dashjr <luke-jr+git@utopios.org>
Date: Sun, 27 Aug 2017 03:56:00 +0000
Subject: [PATCH 1/4] kalarmcal: Update to build with recent Qt4

---
 kalarmcal/kaevent.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kalarmcal/kaevent.cpp b/kalarmcal/kaevent.cpp
index 83dcb12c1..57a383349 100644
--- a/kalarmcal/kaevent.cpp
+++ b/kalarmcal/kaevent.cpp
@@ -2138,7 +2138,7 @@ void KAEventPrivate::setCommandError(KAEvent::CmdErrType error, bool writeConfig
                 case KAEvent::CMD_ERROR_PRE:   errtext = CMD_ERROR_PRE_VALUE;  break;
                 case KAEvent::CMD_ERROR_POST:  errtext = CMD_ERROR_POST_VALUE;  break;
                 case KAEvent::CMD_ERROR_PRE_POST:
-                    errtext = CMD_ERROR_PRE_VALUE + ',' + CMD_ERROR_POST_VALUE;
+                    errtext = CMD_ERROR_PRE_VALUE + QLatin1Char(',') + CMD_ERROR_POST_VALUE;
                     break;
                 default:
                     break;
@@ -2160,7 +2160,7 @@ void KAEvent::setCommandError(const QString& configString)
 void KAEventPrivate::setCommandError(const QString& configString)
 {
     mCommandError = KAEvent::CMD_NO_ERROR;
-    const QStringList errs = configString.split(',');
+    const QStringList errs = configString.split( QLatin1Char(',') );
     if (errs.indexOf(CMD_ERROR_VALUE) >= 0)
         mCommandError = KAEvent::CMD_ERROR;
     else
-- 
2.14.1

