commit 8d194028f484aa85b08d6e47c56aed31f2a66779
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sun Mar 5 18:28:45 2017 +0100

    Adapt to >=gpgme-1.8.0, need for -phtread is gone

diff --git a/cmake/modules/FindGpgme.cmake b/cmake/modules/FindGpgme.cmake
index 5da9a60..76e1bbe 100644
--- a/cmake/modules/FindGpgme.cmake
+++ b/cmake/modules/FindGpgme.cmake
@@ -11,12 +11,12 @@
 #    - GPGME_INCLUDES is the same for all of the above
 #    - GPGME_FOUND is set if any of the above was found
 #  - *nix:
-#    There's also three variants: gpgme{,-pthread,-pth}.
+#    There's also two variants: gpgme{,-pth}.
 #    - The variant used determines the multithreaded use possible:
-#      - gpgme:         no multithreading support available
-#      - gpgme-pthread: multithreading available using POSIX threads
+#      - gpgme:         >=GpgME-1.8.0 multithreading support available
 #      - gpgme-pth:     multithreading available using GNU PTH (cooperative multithreading)
-#    - GPGME_{VANILLA,PTH,PTHREAD}_{FOUND,LIBRARIES} will be set for each of the above
+#    - GPGME_{VANILLA,PTH}_{FOUND,LIBRARIES} will be set for each of the above
+#    - GPGME_PTHREAD_{FOUND,LIBRARIES} will be set for legacy compatibility
 #    - GPGME_INCLUDES is the same for all of the above
 #    - GPGME_FOUND is set if any of the above was found
 #
@@ -156,7 +156,7 @@ else() # not WIN32
   # see WIN32 case for an explanation of what this does:
   set( _seem_to_have_cached_gpgme false )
   if ( GPGME_INCLUDES )
-    if ( GPGME_VANILLA_LIBRARIES OR GPGME_PTHREAD_LIBRARIES OR GPGME_PTH_LIBRARIES )
+    if ( GPGME_VANILLA_LIBRARIES OR GPGME_PTH_LIBRARIES )
       set( _seem_to_have_cached_gpgme true )
     endif()
   endif()
@@ -164,10 +164,10 @@ else() # not WIN32
   if ( _seem_to_have_cached_gpgme )
 
     macro_bool_to_bool( GPGME_VANILLA_LIBRARIES GPGME_VANILLA_FOUND )
-    macro_bool_to_bool( GPGME_PTHREAD_LIBRARIES GPGME_PTHREAD_FOUND )
+    macro_bool_to_bool( GPGME_VANILLA_LIBRARIES GPGME_PTHREAD_FOUND )
     macro_bool_to_bool( GPGME_PTH_LIBRARIES     GPGME_PTH_FOUND     )
 
-    if ( GPGME_VANILLA_FOUND OR GPGME_PTHREAD_FOUND OR GPGME_PTH_FOUND )
+    if ( GPGME_VANILLA_FOUND OR GPGME_PTH_FOUND )
       set( GPGME_FOUND true )
     else()
       set( GPGME_FOUND false )
@@ -189,7 +189,7 @@ else() # not WIN32
 
       exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --version OUTPUT_VARIABLE GPGME_VERSION )
 
-      set( _GPGME_MIN_VERSION "1.1.7" )
+      set( _GPGME_MIN_VERSION "1.7.9" )
       if( ${GPGME_VERSION} VERSION_GREATER ${_GPGME_MIN_VERSION} )
         set( _GPGME_INSTALLED_VERSION_OK TRUE )
       endif()
@@ -207,24 +207,19 @@ else() # not WIN32
 	  set( _gpgme_config_vanilla_libs )
 	endif()
 
-        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --thread=pthread --libs OUTPUT_VARIABLE _gpgme_config_pthread_libs RETURN_VALUE _ret )
-	if ( _ret )
-	  set( _gpgme_config_pthread_libs )
-	endif()
-
         exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --thread=pth     --libs OUTPUT_VARIABLE _gpgme_config_pth_libs     RETURN_VALUE _ret )
 	if ( _ret )
 	  set( _gpgme_config_pth_libs )
 	endif()
 
         # append -lgpg-error to the list of libraries, if necessary
-        foreach ( _flavour vanilla pthread pth )
+        foreach ( _flavour vanilla pth )
           if ( _gpgme_config_${_flavour}_libs AND NOT _gpgme_config_${_flavour}_libs MATCHES "lgpg-error" )
             set( _gpgme_config_${_flavour}_libs "${_gpgme_config_${_flavour}_libs} -lgpg-error" )
           endif()
         endforeach()
 
-        if ( _gpgme_config_vanilla_libs OR _gpgme_config_pthread_libs OR _gpgme_config_pth_libs )
+        if ( _gpgme_config_vanilla_libs OR _gpgme_config_pth_libs )
 
           exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --cflags OUTPUT_VARIABLE _GPGME_CFLAGS )
 
@@ -233,7 +228,7 @@ else() # not WIN32
             string( REGEX REPLACE " *-I"      ";" GPGME_INCLUDES "${_GPGME_CFLAGS}" )
           endif()
 
-          foreach ( _flavour vanilla pthread pth )
+          foreach ( _flavour vanilla pth )
             if ( _gpgme_config_${_flavour}_libs )
 
               set( _gpgme_library_dirs )
@@ -294,10 +289,10 @@ else() # not WIN32
           # This comment above doesn't make sense, the four following lines seem to do nothing. Alex
           set( GPGME_INCLUDES          ${GPGME_INCLUDES} )
           set( GPGME_VANILLA_LIBRARIES ${GPGME_VANILLA_LIBRARIES} )
-          set( GPGME_PTHREAD_LIBRARIES ${GPGME_PTHREAD_LIBRARIES} )
+          set( GPGME_PTHREAD_LIBRARIES ${GPGME_VANILLA_LIBRARIES} )
           set( GPGME_PTH_LIBRARIES     ${GPGME_PTH_LIBRARIES} )
 
-          if ( GPGME_VANILLA_FOUND OR GPGME_PTHREAD_FOUND OR GPGME_PTH_FOUND )
+          if ( GPGME_VANILLA_FOUND OR GPGME_PTH_FOUND )
             set( GPGME_FOUND true )
           else()
             set( GPGME_FOUND false )
@@ -319,7 +314,7 @@ else() # not WIN32
 
   macro_bool_to_01( GPGME_FOUND         HAVE_GPGME         )
   macro_bool_to_01( GPGME_VANILLA_FOUND HAVE_GPGME_VANILLA )
-  macro_bool_to_01( GPGME_PTHREAD_FOUND HAVE_GPGME_PTHREAD )
+  macro_bool_to_01( GPGME_VANILLA_FOUND HAVE_GPGME_PTHREAD )
   macro_bool_to_01( GPGME_PTH_FOUND     HAVE_GPGME_PTH     )
 
 endif() # WIN32 | Unix
@@ -339,16 +334,12 @@ if ( GPGME_QT_FOUND )
   set( _gpgme_flavours "${_gpgme_flavours} Qt" )
 endif()
 
-if ( GPGME_PTHREAD_FOUND )
-  set( _gpgme_flavours "${_gpgme_flavours} pthread" )
-endif()
-
 if ( GPGME_PTH_FOUND )
   set( _gpgme_flavours "${_gpgme_flavours} pth" )
 endif()
 
 # determine the library in one of the found flavours, can be reused e.g. by FindQgpgme.cmake, Alex
-foreach(_currentFlavour vanilla glib qt pth pthread)
+foreach(_currentFlavour vanilla glib qt pth)
    if(NOT GPGME_LIBRARY_DIR)
       get_filename_component(GPGME_LIBRARY_DIR "${_gpgme_${_currentFlavour}_lib}" PATH)
    endif()
diff --git a/cmake/modules/FindQGpgme.cmake b/cmake/modules/FindQGpgme.cmake
index 50013eb..4d94af6 100644
--- a/cmake/modules/FindQGpgme.cmake
+++ b/cmake/modules/FindQGpgme.cmake
@@ -12,13 +12,8 @@ find_package(Gpgme QUIET)
 
 if(GPGME_FOUND)
 
-   if ( WIN32 )
-      find_library(_QGPGME_EXTRA_LIBRARY gpgme++
-                   HINTS ${GPGME_LIBRARY_DIR})
-   else ( WIN32 )
-      find_library(_QGPGME_EXTRA_LIBRARY gpgme++-pthread
-                   HINTS ${GPGME_LIBRARY_DIR})
-   endif()
+   find_library(_QGPGME_EXTRA_LIBRARY gpgme++
+                HINTS ${GPGME_LIBRARY_DIR})
 
    find_library(QGPGME_LIBRARY qgpgme
                 HINTS ${GPGME_LIBRARY_DIR})
@@ -34,7 +29,7 @@ if(GPGME_FOUND)
    if ( WIN32 )
       set(QGPGME_LIBRARIES ${QGPGME_LIBRARY} ${_QGPGME_EXTRA_LIBRARY} ${GPGME_VANILLA_LIBRARIES} ${GPGME_QT_LIBRARIES})
    else ( WIN32 )
-      set(QGPGME_LIBRARIES ${QGPGME_LIBRARY} ${_QGPGME_EXTRA_LIBRARY} ${GPGME_PTHREAD_LIBRARIES})
+      set(QGPGME_LIBRARIES ${QGPGME_LIBRARY} ${_QGPGME_EXTRA_LIBRARY} ${GPGME_VANILLA_LIBRARIES})
    endif()
 
 endif()
