--- a/CMakeLists.txt	2010-01-12 23:27:00.000000000 +0100
+++ b/CMakeLists.txt	2013-12-08 11:50:10.238950313 +0100
@@ -10,13 +10,15 @@
 
 ########### next target ###############
 
-set( kmozillahelper_SRCS
-     main.cpp )
+set(
+	kmozillahelper_SRCS
+	main.cpp
+)
 
 kde4_add_executable( kmozillahelper ${kmozillahelper_SRCS} )
 target_link_libraries( kmozillahelper ${KDE4_KFILE_LIBS} )
 
 # TODO hardcoded path?
-install( TARGETS kmozillahelper DESTINATION /usr/lib/mozilla/ )
+install( TARGETS kmozillahelper DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 install( FILES kmozillahelper.notifyrc  DESTINATION  ${DATA_INSTALL_DIR}/kmozillahelper )
