--- a/wine-1.7.4.ebuild	2013-12-01 11:01:03.000000000 +0100
+++ b/wine-1.7.8.ebuild	2013-12-07 13:57:06.137294261 +0100
@@ -23,7 +23,7 @@
 fi
 
 GV="2.24"
-MV="0.0.8"
+MV="4.5.2"
 PULSE_PATCHES="winepulse-patches-1.7.4"
 WINE_GENTOO="wine-gentoo-2013.06.24"
 DESCRIPTION="Free implementation of Windows(tm) on Unix"
@@ -34,7 +34,7 @@
 		abi_x86_64? ( mirror://sourceforge/${PN}/Wine%20Gecko/${GV}/wine_gecko-${GV}-x86_64.msi )
 	)
 	mono? ( mirror://sourceforge/${PN}/Wine%20Mono/${MV}/wine-mono-${MV}.msi )
-	http://dev.gentoo.org/~tetromino/distfiles/${PN}/${PULSE_PATCHES}.tar.bz2
+	pulseaudio? ( http://dev.gentoo.org/~tetromino/distfiles/${PN}/${PULSE_PATCHES}.tar.bz2 )
 	http://dev.gentoo.org/~tetromino/distfiles/${PN}/${WINE_GENTOO}.tar.bz2"
 
 LICENSE="LGPL-2.1"
@@ -105,7 +105,10 @@
 				app-emulation/emul-linux-x86-baselibs[development]
 				sys-libs/ncurses[abi_x86_32]
 			) )
-			udisks? ( >=app-emulation/emul-linux-x86-baselibs-20130224[development] )
+			udisks? ( || (
+				>=app-emulation/emul-linux-x86-baselibs-20130224[development]
+				sys-apps/dbus[abi_x86_32]
+			) )
 			fontconfig? ( || (
 				app-emulation/emul-linux-x86-xlibs[development]
 				media-libs/fontconfig[abi_x86_32]
@@ -171,7 +174,13 @@
 				>=app-emulation/emul-linux-x86-opengl-20121028[development]
 				media-libs/mesa[osmesa,abi_x86_32]
 			) )
-			xml? ( >=app-emulation/emul-linux-x86-baselibs-20131008[development] )
+			xml? ( || (
+				>=app-emulation/emul-linux-x86-baselibs-20131008[development]
+				(
+					dev-libs/libxml2[abi_x86_32]
+					dev-libs/libxslt[abi_x86_32]
+				)
+			) )
 			scanner? ( app-emulation/emul-linux-x86-medialibs[development] )
 			ssl? ( app-emulation/emul-linux-x86-baselibs[development] )
 			png? ( || (
@@ -257,7 +266,7 @@
 		unpack ${MY_P}.tar.bz2
 	fi
 
-	unpack "${PULSE_PATCHES}.tar.bz2"
+	use pulseaudio && unpack "${PULSE_PATCHES}.tar.bz2"
 	unpack "${WINE_GENTOO}.tar.bz2"
 
 	l10n_find_plocales_changes "${S}/po" "" ".po"
@@ -271,9 +280,9 @@
 		"${FILESDIR}"/${PN}-1.7.2-osmesa-check.patch #429386
 		"${FILESDIR}"/${PN}-1.6-memset-O3.patch #480508
 	)
-	[[ ${PV} == "9999" ]] || PATCHES+=(
-		"../${PULSE_PATCHES}"/*.patch #421365
-	)
+	if [[ ${PV} == "9999" ]] ; then
+		use pulseaudio && PATCHES+="../${PULSE_PATCHES}"/*.patch #421365
+	fi
 
 	autotools-utils_src_prepare
 
@@ -349,7 +358,7 @@
 		$(use_with xml xslt)
 	)
 
-	[[ ${PV} == "9999" ]] || myeconfargs+=( $(use_with pulseaudio pulse) )
+	[[ ${PV} == "9999" ]] || use pulseaudio && myeconfargs+=( $(use_with pulseaudio pulse) )
 
 	if use amd64 && use abi_x86_32; then
 		# Avoid crossdev's i686-pc-linux-gnu-pkg-config if building wine32 on amd64; #472038
