--- a/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp	2014-12-26 13:08:03.943825476 +0100
+++ b/src/Plugins/QtKeyChain/QtKeyChainPassword.cpp	2014-12-26 13:14:24.558067142 +0100
@@ -22,7 +22,11 @@
 
 #include <QString>
 
-#include <qtkeychain/keychain.h>
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+    #include <qtkeychain/keychain.h>
+#else
+    #include <qt5keychain/keychain.h>
+#endif
 
 #include "QtKeyChainPassword.h"
 
