--- drivers/gpu/drm/i915/intel_ringbuffer.c.orig	2012-12-30 17:46:20.037960346 +0100
+++ drivers/gpu/drm/i915/intel_ringbuffer.c	2012-12-30 17:51:28.751100900 +0100
@@ -251,17 +251,16 @@
 {
 	drm_i915_private_t *dev_priv = ring->dev->dev_private;
 	struct drm_i915_gem_object *obj = ring->obj;
-	u32 head;
 
 	/* Stop the ring if it's running. */
 	I915_WRITE_CTL(ring, 0);
+	(void)I915_READ_CTL(ring);
+
 	I915_WRITE_HEAD(ring, 0);
 	ring->write_tail(ring, 0);
 
-	head = I915_READ_HEAD(ring) & HEAD_ADDR;
-
 	/* G45 ring initialization fails to reset head to zero */
-	if (head != 0) {
+	if (I915_READ_HEAD(ring) & HEAD_ADDR) {
 		DRM_DEBUG_KMS("%s head not reset to zero "
 			      "ctl %08x head %08x tail %08x start %08x\n",
 			      ring->name,
