--- a/net-libs/telepathy-glib/telepathy-glib-0.18.1.ebuild	2012-09-03 06:01:22.000000000 +0200
+++ b/net-libs/telepathy-glib/telepathy-glib-0.19.8.ebuild	2012-09-08 19:15:50.037236664 +0200
@@ -23,7 +23,7 @@
 	dev-libs/libxslt
 	virtual/pkgconfig
 	vala? (
-		>=dev-lang/vala-0.14.0:0.14[vapigen]
+		>=dev-lang/vala-0.16.0:0.16[vapigen]
 		>=dev-libs/gobject-introspection-1.30 )"
 
 src_prepare() {
