--- /CMakeLists.txt
+++ /CMakeLists.txt
@@ -153,8 +153,8 @@
 
 # If the KDEPIM_ONLY_KLEO option is true
 if(KDEPIM_ONLY_KLEO)
-  find_package(QGpgme)
-  set_package_properties(QGpgme PROPERTIES DESCRIPTION "The QGpgME library" URL "http://www.kde.org" TYPE REQUIRED PURPOSE "QGpgME is required to build Kleopatra.")
+  find_package(KDE4_QGpgme)
+  set_package_properties(KDE4_QGpgme PROPERTIES DESCRIPTION "The KDE4_QGpgME library" URL "http://www.kde.org" TYPE REQUIRED PURPOSE "KDE4_QGpgME is required to build Kleopatra.")
 
   add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS} ${KDEPIM_DEFINITIONS})
   include_directories (${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES} ${KDEPIMLIBS_INCLUDE_DIRS} ${QT_QTDBUS_INCLUDE_DIR})
@@ -175,8 +175,8 @@
   find_package(ZLIB)
   set_package_properties(ZLIB PROPERTIES DESCRIPTION "The Zlib compression library" URL "http://www.zlib.net" TYPE REQUIRED)
 
-  find_package(QGpgme)
-  set_package_properties(QGpgme PROPERTIES DESCRIPTION "The QGpgMe library" URL "http://www.kde.org" TYPE RECOMMENDED PURPOSE "QGpgME is required to build KMail, KOrganizer and Kleopatra")
+  find_package(KDE4_QGpgme)
+  set_package_properties(KDE4_QGpgme PROPERTIES DESCRIPTION "The KDE4_QGpgME library" URL "http://www.kde.org" TYPE RECOMMENDED PURPOSE "KDE4_QGpgME is required to build KMail, KOrganizer and Kleopatra")
 
   find_package(Grantlee 0.4.0 QUIET CONFIG)
   set_package_properties(Grantlee PROPERTIES DESCRIPTION "The Grantlee Template System" URL "http://www.gitorious.org/grantlee/pages/Home" TYPE REQUIRED PURPOSE "Grantlee is requires for kmail and templating, theming for KJots, KaddressBook, KNotes and MessageViewer(KMail)." )
@@ -253,13 +253,13 @@
   add_subdirectory(messagelist)
   add_subdirectory(templateparser)
 
-  if(QGPGME_FOUND)
+  if(KDE4_QGPGME_FOUND)
     if(Boost_TOPOLOGICAL_SORT_DIR)
       macro_optional_add_subdirectory(kleopatra)
     endif()
   endif()
 
-  # The following components depend on QGpgME.
+  # The following components depend on KDE4_QGpgME.
   add_subdirectory(messageviewer)
   macro_optional_add_subdirectory(messagecomposer)
   add_subdirectory(pimcommon)
--- /examples/etm_usage/CMakeLists.txt
+++ /examples/etm_usage/CMakeLists.txt
@@ -45,7 +45,7 @@
   ${KDEPIMLIBS_AKONADI_LIBS}
   ${KDEPIMLIBS_KABC_LIBS}
   ${KDEPIMLIBS_KMIME_LIBS}
-  ${KDEPIMLIBS_QGPGME_LIBSS}
+  ${KDEPIMLIBS_KDE4_QGPGME_LIBS}
   akonadi_next
   kdepim
   kleo
--- /examples/mailreader/CMakeLists.txt
+++ /examples/mailreader/CMakeLists.txt
@@ -25,7 +25,7 @@
   ${KDE4_KDEUI_LIBS}
   ${KDEPIMLIBS_AKONADI_KMIME_LIBS}
   ${KDEPIMLIBS_AKONADI_LIBS}
-  ${KDEPIMLIBS_QGPGME_LIBS}
+  ${KDEPIMLIBS_KDE4_QGPGME_LIBS}
   akonadi_next
   kdepim
   kleo
--- /kaddressbook/CMakeLists.txt
+++ /kaddressbook/CMakeLists.txt
@@ -28,8 +28,8 @@
 add_subdirectory(icons)
 add_subdirectory(printing)
 add_subdirectory(tests)
-if(QGPGME_FOUND)
-  add_definitions(-DQGPGME_FOUND)
+if(KDE4_QGPGME_FOUND)
+  add_definitions(-DKDE4_QGPGME_FOUND)
 endif()
 
 set(kaddressbook_sendmail_SRCS
@@ -166,8 +166,8 @@
 add_subdirectory(gravatar/autotests)
 add_subdirectory(gravatar/tests)
 
-if(QGPGME_FOUND)
-  target_link_libraries(kaddressbookprivate ${QGPGME_LIBRARIES})
+if(KDE4_QGPGME_FOUND)
+  target_link_libraries(kaddressbookprivate ${KDE4_QGPGME_LIBRARIES})
 endif()
 
 if(PRISON_FOUND)
@@ -200,8 +200,8 @@
   grantleetheme
 )
 
-if(QGPGME_FOUND)
-  target_link_libraries(kaddressbook ${QGPGME_LIBRARIES})
+if(KDE4_QGPGME_FOUND)
+  target_link_libraries(kaddressbook ${KDE4_QGPGME_LIBRARIES})
 endif()
 
 install(TARGETS kaddressbook ${INSTALL_TARGETS_DEFAULT_ARGS})
--- /kaddressbook/xxport/vcard/vcard_xxport.cpp
+++ /kaddressbook/xxport/vcard/vcard_xxport.cpp
@@ -26,12 +26,12 @@
 
 #include <kaddressbookgrantlee/widget/grantleecontactviewer.h>
 
-#ifdef QGPGME_FOUND
+#ifdef KDE4_QGPGME_FOUND
 #include <kde4_gpgme++/context.h>
 #include <kde4_gpgme++/data.h>
 #include <kde4_gpgme++/key.h>
 #include <kde4_qgpgme/dataprovider.h>
-#endif // QGPGME_FOUND
+#endif // KDE4_QGPGME_FOUND
 
 #include <KABC/VCardConverter>
 
@@ -409,7 +409,7 @@
 
 void VCardXXPort::addKey( KABC::Addressee &addr, KABC::Key::Type type ) const
 {
-#ifdef QGPGME_FOUND
+#ifdef KDE4_QGPGME_FOUND
     const QString fingerprint = addr.custom( QLatin1String("KADDRESSBOOK"),
                                              ( type == KABC::Key::PGP ? QLatin1String("OPENPGPFP") : QLatin1String("SMIMEFP") ) );
     if ( fingerprint.isEmpty() ) {
--- /kleopatra/CMakeLists.txt
+++ /kleopatra/CMakeLists.txt
@@ -37,7 +37,7 @@
   ${CMAKE_SOURCE_DIR}/libkdepim
   ${CMAKE_SOURCE_DIR}/mobile/lib
   ${Boost_INCLUDE_DIR}
-  ${QGPGME_INCLUDES}
+  ${KDE4_QGPGME_INCLUDE_DIR}
   ${GPGME_INCLUDES}
 )
 
@@ -367,7 +367,7 @@
 target_link_libraries(kleopatra_bin
   ${_kleopatra_extra_libs}
   kleo
-  ${QGPGME_LIBRARIES}
+  ${KDE4_QGPGME_LIBRARIES}
   ${KDE4_KDEUI_LIBS}
   ${KDEPIMLIBS_KMIME_LIBS}
   ${KDE4_KCMUTILS_LIBS}
--- /kleopatra/kgpgconf/CMakeLists.txt
+++ /kleopatra/kgpgconf/CMakeLists.txt
@@ -21,6 +21,6 @@
 
 kde4_add_executable(kgpgconf ${_kgpgconf_SRCS})
 
-target_link_libraries(kgpgconf ${QGPGME_LIBRARIES} ${KDE4_KDEUI_LIBS})
+target_link_libraries(kgpgconf ${KDE4_QGPGME_LIBRARIES} ${KDE4_KDEUI_LIBS})
 
 install(TARGETS kgpgconf ${INSTALL_TARGETS_DEFAULT_ARGS})
--- /kleopatra/kwatchgnupg/CMakeLists.txt
+++ /kleopatra/kwatchgnupg/CMakeLists.txt
@@ -22,7 +22,7 @@
 target_link_libraries(kwatchgnupg
   ${KDE4_KDEUI_LIBS}
   kleo
-  ${QGPGME_LIBRARIES}
+  ${KDE4_QGPGME_LIBRARIES}
 )
 
 install(TARGETS kwatchgnupg ${INSTALL_TARGETS_DEFAULT_ARGS})
--- /kleopatra/tests/CMakeLists.txt
+++ /kleopatra/tests/CMakeLists.txt
@@ -62,7 +62,7 @@
   ${QT_QTCORE_LIBRARY}
   ${QT_QTGUI_LIBRARY}
   ${KDE4_KDECORE_LIBS}
-  ${QGPGME_LIBRARIES}
+  ${KDE4_QGPGME_LIBRARIES}
 )
 
 ########### next target ###############
@@ -100,13 +100,13 @@
   if(WIN32)
     target_link_libraries(test_uiserver
       ${ASSUAN_VANILLA_LIBRARIES}
-      ${QGPGME_LIBRARIES}
+      ${KDE4_QGPGME_LIBRARIES}
       ws2_32
     )
   else()
     target_link_libraries(test_uiserver
       ${ASSUAN_PTHREAD_LIBRARIES}
-      ${QGPGME_LIBRARIES}
+      ${KDE4_QGPGME_LIBRARIES}
   )
   endif()
 
--- /kmail/CMakeLists.txt
+++ /kmail/CMakeLists.txt
@@ -315,7 +315,7 @@
     ${KDEPIMLIBS_KRESOURCES_LIBS}
     ${KDE4_KCMUTILS_LIBS}
     kleo
-    ${QGPGME_LIBRARIES}
+    ${KDE4_QGPGME_LIBRARIES}
     ksieve
     ksieveui
     kpgp
--- /kmail/kcm_kpimidentities/CMakeLists.txt
+++ /kmail/kcm_kpimidentities/CMakeLists.txt
@@ -17,7 +17,7 @@
 kde4_add_ui_files(kcm_identities_srcs ../ui/identitypage.ui)
 kde4_add_plugin(kcm_kpimidentities ${kcm_identities_srcs})
 target_link_libraries(kcm_kpimidentities
-  ${KDEPIMLIBS_GPGMEPP_LIBS}
+  ${KDEPIMLIBS_KDE4_GPGMEPP_LIBS}
   ${KDEPIMLIBS_AKONADI_LIBS}
   ${KDEPIMLIBS_KPIMIDENTITIES_LIBS}
   ${KDEPIMLIBS_MAILTRANSPORT_LIBS}
--- /kontact/plugins/CMakeLists.txt
+++ /kontact/plugins/CMakeLists.txt
@@ -6,7 +6,7 @@
 )
 
 
-if(QGPGME_FOUND)
+if(KDE4_QGPGME_FOUND)
   if(BUILD_kmail)
     add_subdirectory(kmail)
   endif()
--- /libkleo/CMakeLists.txt
+++ /libkleo/CMakeLists.txt
@@ -104,7 +104,7 @@
 )
 
 set(kleo_LIB_SRCS ${libkleo_core_SRCS} ${libkleo_ui_common_SRCS} ${libkleo_backend_qgpgme_SRCS})
-set(kleo_LIB_LIBS ${KDE4_KDEUI_LIBS} ${QGPGME_LIBRARIES} pimcommon)
+set(kleo_LIB_LIBS ${KDE4_KDEUI_LIBS} ${KDE4_QGPGME_LIBRARIES} pimcommon)
 if ( NOT KDEPIM_ONLY_KLEO )
    set( kleo_LIB_SRCS ${kleo_LIB_SRCS} ${libkleo_backend_chiasmus_SRCS} ${libkleo_ui_SRCS})
    set( kleo_LIB_LIBS ${kleo_LIB_LIBS} ${KDE4_KDECORE_LIBS} ${KDE4_KIO_LIBS})
--- /libkleo/tests/CMakeLists.txt
+++ /libkleo/tests/CMakeLists.txt
@@ -12,7 +12,7 @@
     kleo
     ${KDE4_KDECORE_LIBS}
     ${KDE4_KDEUI_LIBS}
-    ${QGPGME_LIBRARIES}
+    ${KDE4_QGPGME_LIBRARIES}
     ${QT_QTGUI_LIBRARY} )
 endif ()
 
@@ -25,7 +25,7 @@
 
 kde4_add_executable(test_keygen TEST ${test_keygen_SRCS})
 
-target_link_libraries(test_keygen  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QGPGME_LIBRARIES})
+target_link_libraries(test_keygen  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_QGPGME_LIBRARIES})
 
 
 ########### next target ###############
@@ -35,7 +35,7 @@
 
 kde4_add_executable(test_keyselectiondialog TEST ${test_keyselectiondialog_SRCS})
 
-target_link_libraries(test_keyselectiondialog kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QGPGME_LIBRARIES} )
+target_link_libraries(test_keyselectiondialog kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_QGPGME_LIBRARIES} )
 endif ()
 
 ########### next target ###############
@@ -45,7 +45,7 @@
 
 kde4_add_executable(test_cryptoconfig TEST ${test_cryptoconfig_SRCS})
 
-target_link_libraries(test_cryptoconfig  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QGPGME_LIBRARIES})
+target_link_libraries(test_cryptoconfig  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_QGPGME_LIBRARIES})
 
 ########### next target ###############
 
@@ -63,7 +63,7 @@
 
 kde4_add_executable(test_jobs TEST ${test_jobs_SRCS})
 
-target_link_libraries(test_jobs  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QGPGME_LIBRARIES})
+target_link_libraries(test_jobs  kleo ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_QGPGME_LIBRARIES})
 
 ########### next target ###############
 
--- /messagecomposer/CMakeLists.txt
+++ /messagecomposer/CMakeLists.txt
@@ -123,7 +123,7 @@
 
 kde4_add_library( messagecomposer ${LIBRARY_TYPE} ${messagecomposer_src} )
 target_link_libraries( messagecomposer ${KDE4_KIO_LIBS} ${KDEPIMLIBS_KMIME_LIBS} ${KDEPIMLIBS_AKONADI_KMIME_LIBS} ${KDEPIMLIBS_AKONADI_LIBS} ${KDEPIMLIBS_MAILTRANSPORT_LIBS} ${KDEPIMLIBS_KPIMTEXTEDIT_LIBS} ${KDEPIMLIBS_KPIMIDENTITIES_LIBS}
- messagecore messageviewer templateparser kleo kdepim ${QGPGME_LIBRARIES} ${Grantlee_GUI_LIBRARIES} sendlater pimcommon)
+ messagecore messageviewer templateparser kleo kdepim ${KDE4_QGPGME_LIBRARIES} ${Grantlee_GUI_LIBRARIES} sendlater pimcommon)
 
 set_target_properties( messagecomposer PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 
--- /messagecomposer/tests/CMakeLists.txt
+++ /messagecomposer/tests/CMakeLists.txt
@@ -41,7 +41,7 @@
       ${QT_QTGUI_LIBRARY}
       ${QT_QTCORE_LIBRARY}
       ${KDE4_KDEUI_LIBS}
-      ${QGPGME_LIBRARIES}
+      ${KDE4_QGPGME_LIBRARIES}
       )
 endmacro()
 
--- /messageviewer/CMakeLists.txt
+++ /messageviewer/CMakeLists.txt
@@ -206,7 +206,7 @@
   ${KDEPIMLIBS_MAILTRANSPORT_LIBS}
   ${KDE4_KDEUI_LIBS}
   ${KDE4_KPRINTUTILS_LIBS}
-  ${QGPGME_LIBRARIES}
+  ${KDE4_QGPGME_LIBRARIES}
   ${_mailwebview_lib}
   ${Grantlee_CORE_LIBRARIES} 
   ${KDE4_KNEWSTUFF3_LIBS}
--- /messageviewer/tests/CMakeLists.txt
+++ /messageviewer/tests/CMakeLists.txt
@@ -20,7 +20,7 @@
   target_link_libraries( ${_name}
     messageviewer
     kleo
-    ${QGPGME_LIBRARIES}
+    ${KDE4_QGPGME_LIBRARIES}
     ${QT_QTTEST_LIBRARY}
     ${QT_QTCORE_LIBRARY}
     ${QT_QTWEBKIT_LIBRARY}
--- /templateparser/CMakeLists.txt
+++ /templateparser/CMakeLists.txt
@@ -49,7 +49,7 @@
   kleo
   kpgp
   pimcommon
-  ${QGPGME_LIBRARIES}
+  ${KDE4_QGPGME_LIBRARIES}
   ${QT_QTWEBKIT_LIBRARY}
 )
 set_target_properties(templateparser PROPERTIES
--- /templateparser/tests/CMakeLists.txt
+++ /templateparser/tests/CMakeLists.txt
@@ -18,7 +18,7 @@
     kleo
     templateparser
     messageviewer
-    ${QGPGME_LIBRARIES}
+    ${KDE4_QGPGME_LIBRARIES}
     ${QT_QTTEST_LIBRARY}
     ${QT_QTCORE_LIBRARY}
     ${QT_QTWEBKIT_LIBRARY}
