From be726fb31a1acc731f295ea20b5d74384b5f15ba Mon Sep 17 00:00:00 2001
From: Matthieu Imbert <matthieu.imbert@inria.fr>
Date: Sun, 13 Feb 2011 11:28:37 +0100
Subject: [PATCH] drm/i915: Use ACPI OpRegion to determine lid status of eDP

---
 drivers/gpu/drm/i915/intel_dp.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 864417c..2f483b7 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1547,6 +1547,7 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 {
 	struct intel_dp *intel_dp = intel_attached_dp(connector);
 	struct drm_device *dev = intel_dp->base.base.dev;
+	struct drm_i915_private *dev_priv = dev->dev_private;
 	enum drm_connector_status status;
 	struct edid *edid = NULL;
 
@@ -1559,6 +1560,10 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 	if (status != connector_status_connected)
 		return status;
 
+	if (dev_priv->opregion.lid_state)
+		if ( ! ( ioread32(dev_priv->opregion.lid_state) & 0x1 ))
+			return connector_status_disconnected;
+
 	if (intel_dp->force_audio) {
 		intel_dp->has_audio = intel_dp->force_audio > 0;
 	} else {
-- 
1.7.2.3

