diff --git a/include/linux/bio.h b/include/linux/bio.h
index ce33e68..283250a 100644
--- a/include/linux/bio.h
+++ b/include/linux/bio.h
@@ -527,7 +527,7 @@ extern void bio_integrity_init(void);
 #define bioset_integrity_create(a, b)	(0)
 #define bio_integrity_prep(a)		(0)
 #define bio_integrity_enabled(a)	(0)
-#define bio_integrity_clone(a, b, c, d)	(0)
+#define bio_integrity_clone(a, b, c, d)	({0;})
 #define bioset_integrity_free(a)	do { } while (0)
 #define bio_integrity_free(a, b)	do { } while (0)
 #define bio_integrity_endio(a, b)	do { } while (0)
