commit ca6189df285adec6a4cab04e8e5f93bd9798fd4d
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sun Jan 29 19:40:24 2017 +0100

    Fix BUILD_MARBLE_APPS

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8dd98f6..72c2e2c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -53,9 +53,8 @@ endif()
 
 option(BUILD_MARBLE_APPS "Build Marble Desktop/Mobile applications" ON)
 add_feature_info("Marble Desktop/Mobile applications" BUILD_MARBLE_APPS "Build Marble Desktop/Mobile applications. Toggle with BUILD_MARBLE_APPS=YES/NO.")
-if(BUILD_MARBLE_APPS)
-  add_subdirectory( apps )
-endif()
+
+add_subdirectory( apps )
 
 configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/config-marble.h.cmake
diff --git a/src/apps/CMakeLists.txt b/src/apps/CMakeLists.txt
index 9725ae1..7da2eca 100644
--- a/src/apps/CMakeLists.txt
+++ b/src/apps/CMakeLists.txt
@@ -1,11 +1,15 @@
 if (NOT ANDROID)
     add_subdirectory(marble-ui)
+endif()
+if(BUILD_MARBLE_APPS)
+  if (NOT ANDROID)
     add_subdirectory(marble-qt)
     add_subdirectory(marble-mobile)
     add_subdirectory(marble-kde)
     if( QT_QTDECLARATIVE_FOUND OR Qt5Quick_FOUND )
         add_subdirectory(marble-touch)
     endif( QT_QTDECLARATIVE_FOUND OR Qt5Quick_FOUND )
-else()
+  else()
     add_subdirectory(marble_maps)
+  endif()
 endif()
