commit 7a3f6d07de1e1e6e6c8385491c9f798f2e2531cb
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat Jan 28 19:34:14 2017 +0100

    Make MARBLE_NO_WEBKIT available as option

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68b3d78..52fd4f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,7 @@ include( MarbleMacros )
 # Generate the tiles with the tilecreator at compile time
 # if this option is set, srtm.jpg will not be installed but the generated tiles instead
 option(MOBILE "Create a Marble version optimized for handheld devices")
+option(MARBLE_NO_WEBKIT "Create Marble version without Qt4WebKit dependencies" OFF)
 
 ####################################################
 # Specific options for building with Qt 4 vs 5
@@ -140,7 +141,9 @@ else()
   set( QT_USE_QTNETWORK     ON )
   set( QT_USE_QTTEST        ON )
   set( QT_USE_QTSCRIPT      ON )
-  set( QT_USE_QTWEBKIT      ON )
+  if( NOT MARBLE_NO_WEBKIT )
+    set( QT_USE_QTWEBKIT      ON )
+  endif()
   set( QT_USE_QTSVG         ON )
   set( QT_USE_QTDECLARATIVE ON )
   set( QT_USE_QTSQL         ON )
@@ -155,9 +158,6 @@ else()
     FIND_PACKAGE(QtDeclarative)
     include_directories(${QT_QTDECLARATIVE_INCLUDE_DIR})
   endif()
-  if ( NOT QT_QTWEBKIT_FOUND )
-    set ( MARBLE_NO_WEBKIT TRUE )
-  endif()
 endif()
 
 SET(CMAKE_AUTOMOC TRUE)
