commit 756293829694fc5431ca3ef13debc7dee046bbaa
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat Jan 28 23:31:12 2017 +0100

    marble -> marble4

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8fc15dd..4b02188 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-project(marble)
+project(marble4)
 
 ####################################################
 # CMake Settings
diff --git a/FindMarble.cmake.in b/FindMarble.cmake.in
index adcde6b..fedd0c1 100644
--- a/FindMarble.cmake.in
+++ b/FindMarble.cmake.in
@@ -13,7 +13,7 @@ IF ( MARBLE_INCLUDE_DIR AND MARBLE_LIBRARIES )
    SET( MARBLE_FIND_QUIETLY TRUE )
 ENDIF ( MARBLE_INCLUDE_DIR AND MARBLE_LIBRARIES )
 
-FIND_PATH( MARBLE_INCLUDE_DIR NAMES marble/MarbleModel.h )
+FIND_PATH( MARBLE_INCLUDE_DIR NAMES marble4/MarbleModel.h )
 FIND_LIBRARY( MARBLE_LIBRARIES NAMES @MARBLEWIDGET@ )
 
 INCLUDE( FindPackageHandleStandardArgs )
diff --git a/src/lib/marble/CMakeLists.txt b/src/lib/marble/CMakeLists.txt
index 35870f9..451f30b 100644
--- a/src/lib/marble/CMakeLists.txt
+++ b/src/lib/marble/CMakeLists.txt
@@ -613,7 +613,7 @@ install( FILES
     routing/RoutingModel.h
     routing/RoutingProfile.h
 
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/marble
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/marble4
 )
 
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config-phonon.h.cmake
