commit 243137e888daab8b00ddb6e4fc8f21da50e9bd2a
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat Feb 25 14:16:58 2017 +0100

    Switch to KDE Frameworks kwallet

diff --git a/src/common/pwstorage/backend_kwallet.c b/src/common/pwstorage/backend_kwallet.c
index ce7dd72..11548bd 100644
--- a/src/common/pwstorage/backend_kwallet.c
+++ b/src/common/pwstorage/backend_kwallet.c
@@ -66,12 +66,12 @@
 static const gchar *app_id = "darktable";
 static const gchar *kwallet_folder = "darktable credentials";
 
-static const gchar *kwallet_service_name = "org.kde.kwalletd";
-static const gchar *kwallet_path = "/modules/kwalletd";
-static const gchar *kwallet_interface = "org.kde.KWallet";
+static const gchar *kwallet_service_name = "org.kde.kwalletd5";
+static const gchar *kwallet_path = "/modules/kwalletd5";
+static const gchar *kwallet_interface = "kf5_org.kde.KWallet";
 static const gchar *klauncher_service_name = "org.kde.klauncher";
 static const gchar *klauncher_path = "/KLauncher";
-static const gchar *klauncher_interface = "org.kde.KLauncher";
+static const gchar *klauncher_interface = "kf5_org.kde.KLauncher";
 
 // Invalid handle returned by get_wallet_handle().
 static const gint invalid_kwallet_handle = -1;
@@ -147,7 +147,7 @@ static gboolean start_kwallet(backend_kwallet_context_t *context)
   */
   GVariant *ret = g_dbus_connection_call_sync(context->connection, klauncher_service_name, klauncher_path,
                                               klauncher_interface, "start_service_by_desktop_name",
-                                              g_variant_new("(sasassb)", "kwalletd", NULL, NULL, "", FALSE),
+                                              g_variant_new("(sasassb)", "kwalletd5", NULL, NULL, "", FALSE),
                                               NULL, G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
 
   if(check_error(error))
@@ -163,7 +163,7 @@ static gboolean start_kwallet(backend_kwallet_context_t *context)
 
   if(error_string && error_string[0] != '\0')
   {
-    dt_print(DT_DEBUG_PWSTORAGE, "[pwstorage_kwallet] ERROR: error launching kwalletd: %s\n", error_string);
+    dt_print(DT_DEBUG_PWSTORAGE, "[pwstorage_kwallet] ERROR: error launching kwalletd5: %s\n", error_string);
     g_free(error_string);
     return FALSE;
   }
