--- a/CMakeModules/FindOpenGL.cmake	2013-01-01 10:45:47.000000000 +0100
+++ b/CMakeModules/FindOpenGL.cmake	2017-02-11 15:55:30.637336941 +0100
@@ -66,7 +66,7 @@
       /opt/graphics/OpenGL/include
     )
 
-    FIND_LIBRARY(OPENGL_gl_LIBRARY HINTS /usr/lib/x86_64-linux-gnu /usr/lib32
+    FIND_LIBRARY(OPENGL_gl_LIBRARY HINTS /usr/lib/x86_64-linux-gnu /usr/lib
       NAMES GL MesaGL
       PATHS /opt/graphics/OpenGL/lib
             /usr/openwin/lib
@@ -90,7 +90,7 @@
     ENDIF (OPENGL_gl_LIBRARY)
 
     FIND_LIBRARY(OPENGL_glu_LIBRARY
-      NAMES GLU MesaGLU HINTS /usr/lib/x86_64-linux-gnu /usr/lib32
+      NAMES GLU MesaGLU HINTS /usr/lib/x86_64-linux-gnu /usr/lib
       PATHS ${OPENGL_gl_LIBRARY}
             /opt/graphics/OpenGL/lib
             /usr/openwin/lib
