commit 45b66cebc5ec83ada92c2aabbf6d72270237faff
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Thu Jan 5 02:27:36 2017 +0100

    Move tests into tests subdirs

diff --git a/filters/sheets/xlsx/CMakeLists.txt b/filters/sheets/xlsx/CMakeLists.txt
index c3be294..7c5624d 100644
--- a/filters/sheets/xlsx/CMakeLists.txt
+++ b/filters/sheets/xlsx/CMakeLists.txt
@@ -45,15 +45,4 @@ if(SHOULD_BUILD_FILEMANAGER_THUMBNAIL)
     install( FILES sheets_xlsx_thumbnail.desktop DESTINATION ${SERVICES_INSTALL_DIR})
 endif()
 
-########## unit tests ###################
-
-set(TestFormulaParser_SRCS
-    FormulaParser.cpp
-    TestFormulaParser.cpp
-)
-
-ecm_add_test( ${TestFormulaParser_SRCS}
-    TEST_NAME "FormulaParser"
-    NAME_PREFIX "filter-xlsx2ods-"
-    LINK_LIBRARIES komsooxml calligrasheetscommon Qt5::Test
-)
+add_subdirectory( tests )
diff --git a/filters/sheets/xlsx/tests/CMakeLists.txt b/filters/sheets/xlsx/tests/CMakeLists.txt
new file mode 100644
index 0000000..e5ca8ab
--- /dev/null
+++ b/filters/sheets/xlsx/tests/CMakeLists.txt
@@ -0,0 +1,12 @@
+########## unit tests ###################
+
+set(TestFormulaParser_SRCS
+    FormulaParser.cpp
+    TestFormulaParser.cpp
+)
+
+ecm_add_test( ${TestFormulaParser_SRCS}
+    TEST_NAME "FormulaParser"
+    NAME_PREFIX "filter-xlsx2ods-"
+    LINK_LIBRARIES komsooxml calligrasheetscommon Qt5::Test
+)
diff --git a/filters/sheets/xlsx/TestFormulaParser.cpp b/filters/sheets/xlsx/tests/TestFormulaParser.cpp
similarity index 100%
rename from filters/sheets/xlsx/TestFormulaParser.cpp
rename to filters/sheets/xlsx/tests/TestFormulaParser.cpp
diff --git a/filters/sheets/xlsx/TestFormulaParser.h b/filters/sheets/xlsx/tests/TestFormulaParser.h
similarity index 100%
rename from filters/sheets/xlsx/TestFormulaParser.h
rename to filters/sheets/xlsx/tests/TestFormulaParser.h
diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 67c8c14..43efe79 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -3,7 +3,7 @@ include_directories(${KOTEXT_INCLUDES}
                     ${FREETYPE_INCLUDE_DIRS})
 
 add_subdirectory( tests )
-add_subdirectory( styles/tests )
+add_subdirectory( styles )
 
 ########### next target ###############
 set(kotext_LIB_SRCS
diff --git a/libs/text/styles/CMakeLists.txt b/libs/text/styles/CMakeLists.txt
new file mode 100644
index 0000000..46c5585
--- /dev/null
+++ b/libs/text/styles/CMakeLists.txt
@@ -0,0 +1,2 @@
+
+add_subdirectory( tests )
