commit 63a3566d248936ff3aa41dd66a036c18058d2a40
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sat May 23 23:36:32 2015 +0200

    [build] Use cmake-provided FindCups.cmake

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0a45ecf..7521271 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -177,7 +177,7 @@ IF(X11_FOUND)
   LINK_DIRECTORIES( ${XXF86VM_LIBRARY_DIRS} )
 ENDIF(X11_FOUND)
 
-FIND_PACKAGE( CUPS )
+FIND_PACKAGE( Cups )
 IF(HAVE_CUPS)
   INCLUDE_DIRECTORIES( ${CUPS_INCLUDE_DIR} )
   LINK_DIRECTORIES( ${CUPS_LIBRARYS} )

commit dc862388ce9862eca24c548e8d0bb38d0e2b293b
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sun May 24 13:18:20 2015 +0200

    [build] Be able to force system-elektra and -yajl

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4b93fa..c75dfb3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,11 @@ ENDIF()
 FIND_PROGRAM(CMAKE_AR_TMP NAMES ar)
 
 # detect top level projects
-FIND_PACKAGE( Yajl )
+IF(USE_SYSTEM_YAJL)
+  FIND_PACKAGE(Yajl REQUIRED)
+ELSE()
+  FIND_PACKAGE(Yajl)
+ENDIF()
 IF( HAVE_YAJL )
   INCLUDE_DIRECTORIES( ${YAJL_INCLUDE_DIRS} )
   LINK_DIRECTORIES( ${YAJL_LIBRARY_DIRS} )
@@ -73,10 +77,13 @@ ELSE( HAVE_YAJL )
     LINK_DIRECTORIES( ${CMAKE_CURRENT_BINARY_DIR}/yajl )
     SET( YAJL_LIBRARIES yajl-static )
   ENDIF()
-  
 ENDIF( HAVE_YAJL )
 
-FIND_PACKAGE( Elektra )
+IF(USE_SYSTEM_ELEKTRA)
+  FIND_PACKAGE(Elektra REQUIRED)
+ELSE()
+  FIND_PACKAGE(Elektra)
+ENDIF()
 IF(HAVE_ELEKTRA)
   SET( ELEKTRA_SPEC "Requires: elektra
 BuildRequires: elektra-devel" )
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -43,5 +43,5 @@
   ADD_CUSTOM_TARGET( docu ALL
                      DEPENDS html )
 
-  INSTALL( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION share/doc/${PACKAGE_NAME} )
+  INSTALL( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION share/doc/${PACKAGE_NAME}-${OYRANOS_VERSION} )
 ENDIF()
