From c9416a5be30f2c5da8ec0c74a7f1614441fa7ba4 Mon Sep 17 00:00:00 2001
From: Kai-Uwe Behrmann <ku.b@gmx.de>
Date: Wed, 19 Mar 2014 16:52:08 +0100
Subject: [PATCH 1/1] * [core]: fix macro compile warning

---
 src/liboyranos_core/oyranos_threads.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/liboyranos_core/oyranos_threads.c b/src/liboyranos_core/oyranos_threads.c
index b02b32e..b6d8406 100644
--- a/src/liboyranos_core/oyranos_threads.c
+++ b/src/liboyranos_core/oyranos_threads.c
@@ -181,7 +181,7 @@ void oyThreadsInit_(void)
     oyObject_Lock( oy_job_message_list_->oy_, __func__, __LINE__ );
     oyObject_UnLock( oy_job_message_list_->oy_, __func__, __LINE__ );
 
-#ifdef _OPENMP && defined(USE_OPENMP)
+#if defined(_OPENMP) && defined(USE_OPENMP)
     if((omp_get_num_procs() - 1) >= 1)
       oy_thread_count_ = omp_get_num_procs() - 1;
     else
-- 
1.5.6.5
